src/main/java/com/medallia/word2vec/Word2VecModel.java

```diff
@@ -35,15 +33,19 @@
 public class Word2VecModel {
 	final List<String> vocab;
 	final int layerSize;
-	final double[] vectors;
+	final DoubleBuffer vectors;
 	private final static long ONE_GB = 1024 * 1024 * 1024;
 
-	Word2VecModel(Iterable<String> vocab, int layerSize, double[] vectors) {
+	Word2VecModel(Iterable<String> vocab, int layerSize, DoubleBuffer vectors) {
 		this.vocab = ImmutableList.copyOf(vocab);
 		this.layerSize = layerSize;
 		this.vectors = vectors;
 	}
```

Comment:
This is a little bit wrong, but it's as wrong as the original code, so I didn't want to change it.

The problem is that `vectors` is mutable, and this constructor doesn't make a copy. So the caller of the constructor can mess with `vectors` after constructing the object, and it will affect the object.

The proper solution is to duplicate `vectors` here in the constructor, but that would change the semantics of the original code, and might even break existing code, so I didn't want to do that.